File: algorithms/1_easy/C/01880_check_if_word_equals_summation_of_two_words/solution.c
static uint32_t calc_val(const char *restrict s) {
    assert(s != NULL);
    uint32_t res = 0;
    const size_t len = strlen(s);
    assert(len <= 8 && len >= 1);
    for (size_t i = 0; i < len; i += 1) {
        assert(s[i] >= 'a' && s[i] <= 'j');
        res = res * 10 + (uint32_t)(s[i] - 'a');
    }
    return res;
}

bool isSumEqual(const char *first_word, const char *second_word, const char *target_word) {
    const uint32_t f = calc_val(first_word);
    const uint32_t s = calc_val(second_word);
    const uint32_t t = calc_val(target_word);

    return f + s == t;
}

File: algorithms/1_easy/C/01880_check_if_word_equals_summation_of_two_words/solution.rs
impl Solution {
    fn calc_val(bytes: &[u8]) -> u32 {
        let mut res = 0_u32;
        for &b in bytes {
            debug_assert!(b >= b'a' && b <= b'j');
            res = res * 10 + (b - b'a') as u32;
        }
        res
    }

    pub fn is_sum_equal(first_word: String, second_word: String, target_word: String) -> bool {
        assert!(first_word.len() <= 8 && second_word.len() <= 8 && target_word.len() <= 8);

        let f = Self::calc_val(first_word.as_bytes());
        let s = Self::calc_val(second_word.as_bytes());
        let t = Self::calc_val(target_word.as_bytes());

        f + s == t
    }
}
